src/prefect/task_worker.py

```diff
@@ -280,7 +304,9 @@ async def _submit_scheduled_task_run(self, task_run: TaskRun):
                 result_storage=await get_or_create_default_task_scheduling_storage()
             ).update_for_task(task)
             try:
-                run_data = await store.read_parameters(parameters_id)
+                run_data = cast(
+                    dict[str, Any], await store.read_parameters(parameters_id)
+                )
```

Comment:
Question: why the cast?

If it's because read_parameters is yet to be annotated? Then don't cast, just trust that the annotation will come later or fix the annotation there.

If it's because the method returns `Any`, then just annotate the run_data variable:

```
run_data: dict[str, Any] = ...
```